### ISSUE TITLE
VsTest does not insert new parameters

### ISSUE THREAD

**Issue by wasabii** [Microsoft/vsts-tasks#5932]
It seems by design that one can only override existing parameters in *.runsettings. However, I don't see why. Why not allow the user to add whatever parameters he/she wants?

**Comment by acesiddhu** [comment]
yes it is by design today. Adding @PBoraMSFT

**Comment by PBoraMSFT** [comment]
@wasabii - can you share your scenario details on how you intend to use any parameters? The way we designed this feature is that parameters will be accessed via TestContext in the test code (e.g., TestContext.Properties["appURL"] will give you the appURL you may want to connect to in your test for a web-application. If these parameters are not already known, how do you propose to use them in your test code?

**Comment by StefanSchoof** [comment]
I had to run a integration test, that requires username and password. Therefore I created a local runsettings file, which I no committed to git. Since I only need these I first tried without a runsettings, but this not work. Then I added a empty runsettings, because I do not want the username and password in git and someone accidentally add there a real User. But this also does not work, so I add the entry with a empty value. This works, but the setup was more challenging.

**Comment by wasabii** [comment]
So, I don't pull settings from TestContext directly: instead I set up Microsoft.Extensions.Configuration to parse all the available values in it, and inject them into the container. So I tried them as a normal source of configuration values.

Having to specify them all explicitly simply becomes a burden. I would rather leave it up to the build on what it wants to set. I have large sets of settings, like, some related to Selenium tests. Fake user accounts in my suite. Fake connection strings. Lots of things.

It's really just a matter of not wanting to list them all out.
